apps/src/javalab/JavalabFileHelper.js

```diff
@@ -18,20 +18,80 @@ export function getDefaultFileContents(filename, viewMode) {
 export const getTabKey = index => `file-${index}`;
 
 export const fileMetadataForEditor = (sources, isEditingStartSources) => {
+  /*
+  'sources' contains file information as key-value pairs.
+  The key is the file name such as 'Class1.java'.
+  The value is a file object with the keys: text (source code),
+  tabOrder (order of the file tab in orderedTabs from 0 to (number of files - 1)),
+  isVisible, and isValidation (default is false)
+  The orderOfFiles array will contain the 'fileOrder' of each file as they are assigned
+  their tabKey names (such as 'file-0', 'file-1') which is stored in fileMetadata.
+  For example, if sources contains the following:
+  {
+    'Class1.java': {text: '', tabOrder: 2, isVisible: true},
+    'Class2.java': {text: '', tabOrder: 0, isVisible: true},
+    'Class3.java': {text: '', tabOrder: 1, isVisible: true}
```

Comment:
Did you consider having the file order in the project.json change rather than adding a new key? It seems like that approach might be more straightforward rather than having to assign metadata and orderedTabKeys after the fact.

Reply:
I did consider changing the order in the json file, but I think that the order of object keys is not guaranteed. Also, when in levelbuilder startMode, validation and support files are added to sources for the levelbuilder and in the corresponding level for the student, these validation and support files are not included, but the other files would remain in the order set by the levelbuilder.

Reply:
Yes, if the files were stored in an array rather than an object, the logic to handle the ordering would have been more simple since order would've been guaranteed and we could rearrange order in the array when saving to sources. But it seems like this would require refactoring of other parts of Java Lab.